meson.build

```diff
@@ -48,7 +48,22 @@ if enable_nls
 endif
 
 conf.set('HAVE_LOCALE_T', enable_locale and cc.has_header_symbol('locale.h', 'locale_t'))
-conf.set('HAVE_ICONV', enable_locale and cc.has_function('iconv'))
+if meson.version().version_compare('>= 0.60')
+  iconv_dep = dependency('iconv', required: false)
+else
+  iconv_dep = cc.find_library('iconv', required: false)
+  message('linking iconv snippet...')
+  iconv_linking_test = cc.links('''#include <iconv.h>
+    int main() {
+    iconv_open("","");
+    }''', dependencies: iconv_dep)
+  if iconv_linking_test
+    message('iconv linked successfully')
+  else
+    message('iconv linking test failed')
+  endif
+endif
+conf.set('HAVE_ICONV', enable_locale and cc.has_function('iconv', dependencies: iconv_dep))
```

Comment:
You unconditionally add iconv_dep even if there is both a (working) builtin version and an external copy of the GNU one. You also do a bunch of checks without first attempting to see if it works OOTB.